### PR TITLE
feat: add experimental support for reverse scans public preview

### PR DESCRIPTION
**Pull request by igorbernstein2**

### CODE REVIEW

**Review by mutianf — APPROVED** [review]

**Comment by averikitsch** [comment]
@igorbernstein2 do you have an ETA on the completion of this PR?

**Review by mutianf — APPROVED** [review]

File: bigtable/hbase/snippets/src/main/java/com/example/bigtable/Reads.java [hunk 1]
```diff
@@ -237,6 +236,38 @@ public static void readPrefix(String projectId, String instanceId, String tableI
...
+          new Scan()
+              .setReversed(true)
+              .setLimit(2)
+              .withStartRow(Bytes.toBytes("phone#4c410523#20190505"));
+      ResultScanner rows = table.getScanner(revScan);
+
+      for (Result row : rows) {
+        printRow(row);
+      }
+    } catch (IOException e) {
```
Comment (billyjacobson) [inline bigtable/hbase/snippets/src/main/java/com/example/bigtable/Reads.java:264]: can you get rid of the try catch and just let the method throw the exception 

https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md#exception-handling

**Review by billyjacobson — APPROVED** [review]
looks good

File: bigtable/hbase/snippets/src/main/java/com/example/bigtable/Reads.java (hunk 1 above)
Comment (igorbernstein2) [reply bigtable/hbase/snippets/src/main/java/com/example/bigtable/Reads.java:264]: I think that should come separately, to keep the structure consistent with the rest of the file.

Comment (billyjacobson) [reply bigtable/hbase/snippets/src/main/java/com/example/bigtable/Reads.java:264]: – will refactor in a larger PR

File: bigtable/hbase/snippets/pom.xml [hunk 2]
```diff
@@ -62,7 +62,7 @@
     <dependency>
       <groupId>com.google.cloud.bigtable</groupId>
       <artifactId>bigtable-hbase-1.x</artifactId>
-      <version>2.9.0</version>
+      <version>2.10.0</version>
```
Comment (Sita04) [inline bigtable/hbase/snippets/pom.xml:65]: Do you need to manually set this version? Can it be pulled from the bom?

File: bigtable/hbase/snippets/src/main/java/com/example/bigtable/Reads.java [hunk 3]
```diff
@@ -237,6 +236,38 @@ public static void readPrefix(String projectId, String instanceId, String tableI
...
+  // [START bigtable_reverse_scan_hbase]
+  public static void readRowsReversed() {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "my-project-id";
+    String instanceId = "my-instance-id";
+    String tableId = "mobile-time-series";
+    readRowsReversed(projectId, instanceId, tableId);
+  }
+
+  public static void readRowsReversed(String projectId, String instanceId, String tableId) {
```
Comment (Sita04) [inline bigtable/hbase/snippets/src/main/java/com/example/bigtable/Reads.java:248]: Kindly add a method comment here.

File: bigtable/hbase/snippets/src/main/java/com/example/bigtable/Reads.java [hunk 4]
```diff
@@ -237,6 +236,38 @@ public static void readPrefix(String projectId, String instanceId, String tableI
...
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
```
Comment (Sita04) [inline bigtable/hbase/snippets/src/main/java/com/example/bigtable/Reads.java:251]:
```suggestion
    // Initialize client that will be used to send requests. This client only needs to be created
    // once, and can be reused for multiple requests.
```

File: bigtable/hbase/snippets/src/main/java/com/example/bigtable/Reads.java (hunk 1 above)
Comment (Sita04) [reply bigtable/hbase/snippets/src/main/java/com/example/bigtable/Reads.java:264]: +1 to that. If we are not demonstrating what to do with the exception, it should be thrown.

**Review by Sita04 — CHANGES_REQUESTED** [review]

File: bigtable/hbase/snippets/pom.xml (hunk 2 above)
Comment (igorbernstein2) [reply bigtable/hbase/snippets/pom.xml:65]: These version shouldnt be here to begin with. I will remove them in a follow up PR, but I didnt want to expand scope here

**Review by Sita04 — APPROVED** [review]

**Pull request merged**